freqtrade/freqtradebot.py

```diff
@@ -470,17 +464,32 @@ def check_and_call_adjust_trade_position(self, trade: Trade):
         stake_amount = strategy_safe_wrapper(self.strategy.adjust_trade_position,
                                              default_retval=None)(
             trade=trade, current_time=datetime.now(timezone.utc), current_rate=current_rate,
-            current_profit=current_profit, min_stake=min_stake_amount, max_stake=max_stake_amount)
+            current_profit=current_profit, min_stake=min_stake_amount,
+            max_stake=max_stake_amount, current_entry_rate=current_entry_rate,
+            current_exit_rate=current_exit_rate)
 
         if stake_amount is not None and stake_amount > 0.0:
             # We should increase our position
+            if self.strategy.max_entry_position_adjustment > -1:
+                count_of_buys = trade.nr_of_successful_buys
+                if count_of_buys > self.strategy.max_entry_position_adjustment:
+                    logger.debug(f"Max adjustment entries for {trade.pair} has been reached.")
+                    return
+                else:
+                    logger.debug("Max adjustment entries is set to unlimited.")
             self.execute_entry(trade.pair, stake_amount, trade=trade)
 
         if stake_amount is not None and stake_amount < 0.0:
             # We should decrease our position
-            # TODO: Selling part of the trade not implemented yet.
-            logger.error(f"Unable to decrease trade position / sell partially"
-                         f" for pair {trade.pair}, feature not implemented.")
+            amount = abs(stake_amount) / current_exit_rate
```

Comment:
Strategy should return value as Decimal. Else 
```
a = Decimal(float(Decimal(-13.533) * Decimal(.3134))) / Decimal(.3134)
print(float(a))
>>> -13.533000000000001
```

Reply:
Should we use this function?
```
def div(a, b):
    p = Decimal(str(a)).as_tuple().exponent
    return round(float(Decimal(a) / Decimal(b)), -p)
```

Reply:
Should we need this function?
```
def div(a, b):
    p = Decimal(str(b)).as_tuple().exponent
    return round(float(Decimal(a) / Decimal(b)), -p)
 ```
 
 FYI, price_to_precision not works for `FEG/USDT` as it has price in 12 decimal points, the function truncates to 10 digit.